v21.1/performance-recipes-solutions.md

```diff
@@ -0,0 +1,84 @@
+    <button class="filter-button page-level" data-scope="fullscans">Full table scans<strong></strong></button>
+    <button class="filter-button page-level" data-scope="primarykey">Suboptimal primary keys<strong></strong></button>
+</div>
+
+<section class="filter-content" markdown="1" data-scope="contention">
+## Indicators that your workoad is experiencing contention
+
+Your application is experiencing degraded performance with serialization errors like:
+
+  - `SQLSTATE: 40001`
+  - `RETRY_WRITE_TOO_OLD`
+  - `RETRY_SERIALIZABLE`
+
+The [SQL Statement Contention graph](ui-sql-dashboard.html#sql-statement-contention) graph is showing spikes over time.
+
+{%comment%} Add screenshot of high-contention graph {%endcomment%}
+
+The [SQL Statement Errors graph](ui-sql-dashboard.html#sql-statement-errors) graph is showing spikes in retries over time.
+
+{%comment%} Add screenshot of high-retries graph {%endcomment%}
+
+## Fixing contention problems
+
+{% include {{ page.version.version }}/performance/statement-contention.md %}
+</section>
+
+<section class="filter-content" markdown="1" data-scope="fullscans">
+
+## Indicators that your workload has statements with full table scans
+
+The following query returns statements with full table scans in their statement plan:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SHOW FULL TABLE SCANS;
+~~~
+
+The following query against the `CRDB_INTERNAL.node_statement_statistics` table returns results:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT count(*) as total_full_scans
+FROM crdb_internal.node_statement_statistics
+WHERE FullTableScan = 'True';
+~~~
+
+Viewing the statement plan on the [Statement details page](ui-statements-page.html#statement-details-page) of the DB Console indicates that the plan contains full table scans.
+
+If the statement plans returned by the [`EXPLAIN`](sql-tuning-with-explain.html) and [`EXPLAIN ANALYZE` commands](explain-analyze.html) indicate that there are full table scans.
+
+## Fixing full table scans in statements
+
+Full table scans often result in poor statement performance. Not every full table scan is an indicator of poor performance, however. The cost-based optimizer may decide on a full table scan when other index or join scans would result in longer execution time.
+
+[Examine the statements](sql-tuning-with-explain.html) that result in full table scans and consider adding [secondary indexes](schema-design-indexes.html#create-a-secondary-index).
+
+</section>
+
+<section class="filter-content" markdown="1" data-scope="primarykey">
+
+## Indicators that your tables are using suboptimal primary keys
+
+The [Hardware metrics dashboard](ui-hardware-dashboard.html) in the DB Console shows high resource usage.
+
+The Problem Ranges report on the [Advanced Debug page](ui-debug-pages.html) of the DB Console indicates a high number of queries per second.
+
```

Comment:
Same comment as below. Perhaps we should call out per node? This scenario would lead to hot spots/nodes.